include/pybind11/pybind11.h

```diff
@@ -795,11 +795,17 @@ class module : public object {
         if (!options::show_user_defined_docstrings()) doc = nullptr;
 #if PY_MAJOR_VERSION >= 3
         PyModuleDef *def = new PyModuleDef();
+        // Ensure that Python does not try to delete this.
+        Py_INCREF(def);
```

Comment:
What's happening here? Why would the python allocator be trying to free memory that it has never allocated?

Reply:
Sorry, this was me trying a theory on Travis, because I could not reproduce the failure locally even in Docker

Reply:
(My theory was wrong, anyway)

Reply:
Oh, I didn't notice the travis failure before. I can still repro, though I get the same error as travis, but on pytest, not cpptest.

Hop over to [gitter](https://gitter.im/pybind/Lobby). It will be easier to chat there and look into this.